addons/utm/models/utm_object_mixin.py

```diff
@@ -0,0 +1,62 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import re
+import uuid
+
+from odoo import fields, models, api
+
+
+class UtmObject(models.AbstractModel):
+    """Mixin class for UTM objects (source, medium, campaign...).
+
+    Store all the common fields/behavior of the UTM objects in this class (like the name,
+    the identifier, the generation of the identifier...).
+    """
+
+    _name = 'utm.object'
```

Comment:
Adding a mixin is adding complexity ... Indeed you already have the name field hidden on mailing model as it comes from its inherit on utm object. Adding another layer adds complexity when reading code.

I would instead embed name and identifier on each main models. Then utm.mixin could offer those _generate_name_from_identifier and _generate_name_from_identifier generic methods, or available as standalone tools taking an env.

Reply:
@tde-banana-odoo But so it means that I will also need to override the create method in utm-campaign,source,medium  ? :/
(to generate the identifier from the name)

Reply:
Indeed.